pysat/_instrument.py

```diff
@@ -1122,7 +1122,9 @@ def next(self, verifyPad=False):
         if self._iter_type == 'date':
             if self.date is not None:
                 idx, = np.where(self._iter_list == self.date)
-                if (len(idx) == 0) | (idx+1 >= len(self._iter_list)):
+                if len(idx) == 0:
+                    raise StopIteration('Outside the set date boundaries.')
+                if idx[-1]+1 >= len(self._iter_list):
```

Comment:
Should these two if statements be combined?  The error message is the same.  If not, I recommend making the error messages slightly different.

Reply:
If idx=[] then idx[-1] produces an error so both checks can't be in the same line. I switched line 1127 to an elif to prevent said error, though the raise prevents it anyway. I updated the error text for the first case to be more meaningful.